src/lib.rs

```diff
@@ -1320,6 +1347,62 @@ pub async fn blocktank_wipe_all() -> Result<(), BlocktankError> {
     }).await.unwrap()
 }
 
+#[uniffi::export]
+pub async fn upsert_info(info: IBtInfo) -> Result<(), BlocktankError> {
+    let rt = ensure_runtime();
+    rt.spawn(async move {
+        let cell = ASYNC_DB.get().ok_or(BlocktankError::ConnectionError {
+            error_details: "Database not initialized. Call init_db first.".to_string()
+        })?;
+        let guard = cell.lock().await;
+        let db = guard.blocktank_db.as_ref().ok_or(BlocktankError::ConnectionError {
+            error_details: "Database not initialized. Call init_db first.".to_string()
+        })?;
+        let external_info: rust_blocktank_client::IBtInfo = info.into();
+        db.upsert_info(&external_info).await
+    }).await.unwrap_or_else(|e| Err(BlocktankError::ConnectionError {
+        error_details: format!("Runtime error: {}", e)
+    }))
+}
```

Comment:
is this basically just inserting?
Not sure `upsert` is very accurate of a name here, because we don't have an identifier… but then again, it's not really an `insert` if it replaces the existing `Info` in DB.

Maybe `setInfo` or `updateInfo`?! 🤔  I would prefer `set` because, since `Info` can be null, then when null we don't update, we set it.